File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/dependecyinjection/UtilsModule.java
package com.baytrendy.shopifyapp.dependecyinjection;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000P\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0002\b\u0006\b\u0007\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u0015\u0010\t\u001a\u00020\n2\u0006\u0010\u000b\u001a\u00020\fH\u0001\u00a2\u0006\u0002\b\rJ\u0015\u0010\u000e\u001a\u00020\u000f2\u0006\u0010\u0002\u001a\u00020\u0003H\u0001\u00a2\u0006\u0002\b\u0010J\u001d\u0010\u0011\u001a\u00020\u00122\u0006\u0010\u0002\u001a\u00020\u00032\u0006\u0010\u0013\u001a\u00020\u0006H\u0001\u00a2\u0006\u0002\b\u0014J%\u0010\u0015\u001a\u00020\u00162\u0006\u0010\u0017\u001a\u00020\n2\u0006\u0010\u0018\u001a\u00020\u000f2\u0006\u0010\u0013\u001a\u00020\u0012H\u0001\u00a2\u0006\u0002\b\u0019J\u0015\u0010\u001a\u001a\u00020\u001b2\u0006\u0010\u001c\u001a\u00020\u0016H\u0001\u00a2\u0006\u0002\b\u001dJ\r\u0010\u001e\u001a\u00020\u0003H\u0001\u00a2\u0006\u0002\b\u001fJ\r\u0010 \u001a\u00020!H\u0001\u00a2\u0006\u0002\b\"J\u001d\u0010#\u001a\u00020\f2\u0006\u0010$\u001a\u00020!2\u0006\u0010%\u001a\u00020\u0006H\u0001\u00a2\u0006\u0002\b&R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0014\u0010\u0005\u001a\u00020\u00068AX\u0080\u0004\u00a2\u0006\u0006\u001a\u0004\b\u0007\u0010\b\u00a8\u0006\'"}, d2 = {"Lcom/shopify/shopifyapp/dependecyinjection/UtilsModule;", "", "context", "Landroid/content/Context;", "(Landroid/content/Context;)V", "requestHeader", "Lokhttp3/OkHttpClient;", "getRequestHeader$base_debug", "()Lokhttp3/OkHttpClient;", "getApiCallInterface", "Lcom/shopify/shopifyapp/utils/ApiCallInterface;", "retrofit", "Lretrofit2/Retrofit;", "getApiCallInterface$base_debug", "getAppDatabase", "Lcom/shopify/shopifyapp/dbconnection/database/AppDatabase;", "getAppDatabase$base_debug", "getGraphClient", "Lcom/shopify/buy3/GraphClient;", "client", "getGraphClient$base_debug", "getRepository", "Lcom/shopify/shopifyapp/repositories/Repository;", "apiCallInterface", "appDatabase", "getRepository$base_debug", "getViewModelFactory", "Landroidx/lifecycle/ViewModelProvider$Factory;", "myRepository", "getViewModelFactory$base_debug", "provideContext", "provideContext$base_debug", "provideGson", "Lcom/google/gson/Gson;", "provideGson$base_debug", "provideRetrofit", "gson", "okHttpClient", "provideRetrofit$base_debug", "base_debug"})
@dagger.Module()
public final class UtilsModule {
    private final android.content.Context context = null;
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final okhttp3.OkHttpClient getRequestHeader$base_debug() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final com.google.gson.Gson provideGson$base_debug() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final retrofit2.Retrofit provideRetrofit$base_debug(@org.jetbrains.annotations.NotNull()
    com.google.gson.Gson gson, @org.jetbrains.annotations.NotNull()
    okhttp3.OkHttpClient okHttpClient) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final com.baytrendy.shopifyapp.utils.ApiCallInterface getApiCallInterface$base_debug(@org.jetbrains.annotations.NotNull()
    retrofit2.Retrofit retrofit) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final com.baytrendy.shopifyapp.repositories.Repository getRepository$base_debug(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.utils.ApiCallInterface apiCallInterface, @org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.dbconnection.database.AppDatabase appDatabase, @org.jetbrains.annotations.NotNull()
    com.shopify.buy3.GraphClient client) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final androidx.lifecycle.ViewModelProvider.Factory getViewModelFactory$base_debug(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.repositories.Repository myRepository) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final android.content.Context provideContext$base_debug() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final com.baytrendy.shopifyapp.dbconnection.database.AppDatabase getAppDatabase$base_debug(@org.jetbrains.annotations.NotNull()
    android.content.Context context) {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Singleton()
    @dagger.Provides()
    public final com.shopify.buy3.GraphClient getGraphClient$base_debug(@org.jetbrains.annotations.NotNull()
    android.content.Context context, @org.jetbrains.annotations.NotNull()
    okhttp3.OkHttpClient client) {
        return null;
    }
    
    public UtilsModule(@org.jetbrains.annotations.NotNull()
    android.content.Context context) {
        super();
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/dependecyinjection/UtilsModule.kt
package com.baytrendy.shopifyapp.dependecyinjection

import android.content.Context
import android.util.Log
import androidx.lifecycle.ViewModelProvider
import androidx.room.Room
import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import com.shopify.buy3.GraphClient
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.dbconnection.database.AppDatabase
import com.baytrendy.shopifyapp.repositories.Repository
import com.baytrendy.shopifyapp.utils.ApiCallInterface
import com.baytrendy.shopifyapp.utils.Constant
import com.baytrendy.shopifyapp.utils.Urls
import com.baytrendy.shopifyapp.utils.ViewModelFactory
import java.util.concurrent.TimeUnit
import javax.inject.Singleton
import dagger.Module
import dagger.Provides
import me.jessyan.retrofiturlmanager.RetrofitUrlManager
import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import org.json.JSONObject
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.net.URI

@Module
class UtilsModule(private val context: Context) {

    internal val requestHeader: OkHttpClient
        @Provides
        @Singleton
        get() {
            val httpClient = RetrofitUrlManager.getInstance().with(OkHttpClient.Builder())
            httpClient.addInterceptor { chain ->
                val original = chain.request()
                val request = original.newBuilder().build()
                Log.i("OkHttp", " " + request.url())
                Log.i("OkHttp", " " + request.headers())
                Log.i("OkHttp", " " + request.method())
                chain.proceed(request)
            }
                    .connectTimeout(100, TimeUnit.SECONDS)
                    .writeTimeout(100, TimeUnit.SECONDS)
                    .readTimeout(300, TimeUnit.SECONDS)

            return httpClient.build()
        }

    @Provides
    @Singleton
    internal fun provideGson(): Gson {
        val builder = GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
        return builder.setLenient().create()
    }

    @Provides
    @Singleton
    internal fun provideRetrofit(gson: Gson, okHttpClient: OkHttpClient): Retrofit {


        return Retrofit.Builder()
                .baseUrl(Urls.BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build()
    }

    @Provides
    @Singleton
    internal fun getApiCallInterface(retrofit: Retrofit): ApiCallInterface {
        return retrofit.create(ApiCallInterface::class.java)
    }

    @Provides
    @Singleton
    internal fun getRepository(apiCallInterface: ApiCallInterface, appDatabase: AppDatabase, client: GraphClient): Repository {
        return Repository(apiCallInterface, appDatabase, client)
    }

    @Provides
    @Singleton
    internal fun getViewModelFactory(myRepository: Repository): ViewModelProvider.Factory {
        return ViewModelFactory(myRepository)
    }

    @Provides
    @Singleton
    internal fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    internal fun getAppDatabase(context: Context): AppDatabase {
        return Room.databaseBuilder(context, AppDatabase::class.java, "MageNative").fallbackToDestructiveMigration().allowMainThreadQueries().build()
    }

    @Provides
    @Singleton
    internal fun getGraphClient(context: Context, client: OkHttpClient): GraphClient {

        return GraphClient.build(context, context.resources.getString(R.string.shopdomain), context.resources.getString(R.string.apikey)
        ) {
            httpClient = client
            httpCache(context.cacheDir, {
                cacheMaxSizeBytes = 1024 * 1024 * 10
                defaultCachePolicy = Constant.policy
                Unit
            })
            Unit
        }
    }
}
